adapter/serviceregistry/aggregate/controller.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggregate
+
+import (
+	"github.com/golang/glog"
+
+	"istio.io/pilot/model"
+	"istio.io/pilot/platform"
+)
+
+// Registry specifies the collection of service registry related interfaces
+type Registry interface {
+	model.Controller
+	model.ServiceDiscovery
+	model.ServiceAccounts
+}
+
+// Controller aggregates data across different registries and monitors for changes
+type Controller struct {
+	registries map[platform.ServiceRegistry]Registry
+	regOrder   []platform.ServiceRegistry
```

Comment:
why not just a list of []Registry?

Reply:
This is for future cache optimization.. Having an idea of the underlying platform helps to invalidate the platform specific cache and also take any platform specific actions..

Reply:
Still, add name property to registry, and no need to have an extra map.